.github/workflows/e2e-test.yml

```diff
@@ -46,16 +46,31 @@ jobs:
           conda install -c conda-forge mamba
           mamba env update --file conda-environment.yml --name base
           pip install -e .
+
+      - name: Cache UI dependencies
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/Cypress
+            ui/node_modules
+          key: cache-${{ runner.os }}-${{ hashFiles('ui/package-lock.json') }}
+
+      - name: Install UI dependencies
+        run: npm install --prefix ./ui
+
       - name: Build UI
-        run: |
-          npm install --prefix ./ui 
-          npm run --prefix ./ui build
+        run: npm run --prefix ./ui build
         env:
           DISABLE_ESLINT_PLUGIN: true
-      - name: Run mxcube backend
-        run: |
-          mxcubeweb-server -r ./test/HardwareObjectsMockup.xml/ --static-folder $(pwd)/ui/build/ -L debug &
-      - name: Cypress run
-        uses: cypress-io/github-action@v5
```

Comment:
This action was also installing dependencies. In fact, it's quite the behemoth: it does the caching, the installation, can even run a dev server ... I prefer to make all these steps explicit for maintainability (e.g. won't have to remember to configure it properly if we switch to `pnpm` in the future, for instance).